### ISSUE TITLE
Unexpected error occurs when trying to install a package having invalid client policies in PM UI 

### ISSUE THREAD

**Issue by v-longmin** [NuGet/Home#10034]
## Details about Problem
VS Version: Main\30515.165
OS Version: Windows-10-Enterprise-RS5

## Installations
1. Install VS: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4068483&view=ms.vss-build-web.run-extensions-tab
## Detailed repro steps 
1. Create a new test certificate` .\CreateTestCertificate.ps1 –AddAsTrustedRootAuthority.`
2. Sign a package: `NuGet.exe sign <PackageFilePath> -CertificatePath`.
3. Create a project, add required mode in NuGet.config file with no trusted signers:
  ```
<config>
     <add key="signatureValidationMode" value="require" />
  </config>
```
4. Reload the solution, update the package source to above package path, and install the package into a project in PM UI/CLI.

## Expected
Only error NU3034: signatureValidationMode is set to require, so packages are allowed only if signed by trusted signers; however, no trusted signers were specified occurred.

## Actual
Unexpected error occured below screenshot.
![image](https://user-images.githubusercontent.com/40230425/93317206-c4680180-f83f-11ea-9f18-14d1a32855ac.png)

## Error Message

Severity | Code | Description | Project | File | Line | Suppression State
-- | -- | -- | -- | -- | -- | --
Error |   | StreamJsonRpc.RemoteInvocationException:   Package restore failed. Rolling back package changes for 'PC01'.   at   Microsoft.ServiceHub.Framework.ServiceJsonRpcDescriptor.LocalProxyGeneration.<ReturnedValueTaskHelper>d__24.MoveNext()---   End of stack trace from previous location where exception was thrown ---   at   System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task   task)   at   System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task   task)   at System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()   at   NuGet.PackageManagement.UI.UIActionEngine.<>c__DisplayClass11_0.<<PerformActionImplAsync>b__0>d.MoveNext() |   |   |   |  
Error |   | Package restore   failed. Rolling back package changes for 'PC01'. |   |   |   |  
Error | NU3034 | Package 'Package1   1.0.0' from source 'C:\Users\v-minlon\source\repos\Package1\Package1': This   package is signed but not by a trusted signer. | PC01 | C:\Users\v-minlon\source\repos\PC01\PC01\PC01.csproj | 1 |  
Warning | NU3027 | Package 'Package1   1.0.0' from source 'C:\Users\v-minlon\source\repos\Package1\Package1': The   signature should be timestamped to enable long-term signature validity after   the certificate has expired. | PC01 | C:\Users\v-minlon\source\repos\PC01\PC01\PC01.csproj | 1 |  

## Note
1. This issue repros in all the scenarios expected to have a error. 
2. It start to repro from Dev\5.8.0.6836.

**Comment by dtivel** [comment]
In 16.8 we introduced multiple new VS OE services.  We basically moved a bunch of existing code behind some new, remoteable interfaces.  While these VS OE services can be local or remote, client code is generally blind to whether a service is actually local or remote.  This is great because client code doesn't need different code paths for local and remote scenarios.

Before adding VS OE support, NuGet client code would call some `NuGetPackageManager` method to a install (or uninstall) package, and if that failed, client code would catch a specific exception (usually a NuGet-specific exception, like `NuGet.Packaging.Signing.SignatureException`) which more aptly described the failure.  In displaying the exception to the user within Visual Studio, [we log different things depending on the exception type](https://github.com/NuGet/NuGet.Client/blob/ba87e7e45851add0e46a217c14fa2c95973244a9/src/NuGet.Clients/NuGet.PackageManagement.UI/UserInterfaceService/NuGetUI.cs#L358-L405).  If it was an unexpected type, we'd fallback to `Exception.ToString()`.  Also, although we sometimes add duplicate entries to the Visual Studio Error List, Visual Studio automatically deduplicates entries based on their message ([`ITableEntry.Identity`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.shell.tablemanager.itableentry.identity?view=visualstudiosdk-2019#Microsoft_VisualStudio_Shell_TableManager_ITableEntry_Identity)), so you ultimately wouldn't see duplicate entries in the Error List.

After adding VS OE support, this same `NuGetPackageManager` call is now hidden behind a VS OE service; when the corresponding service call fails, client code always catches a `StreamJsonRpc.RemoteInvocationException`.  This is true for every one of our Nexus services:  where an exception could have been thrown before, client code will now only get a `RemoteInvocationException`.  Because `RemoteInvocationException` is not one of the special exceptions we handle, we log `Exception.ToString()` by default, which generates a message essentially with the format `<ExceptionTypeName>: <ExceptionMessage><ExceptionCallStack>` instead of just `<ExceptionMessage>`.  These two messages are different so there's nothing to deduplicate.  This is why we see 2 very similar errors in the Error List:
```
Package restore failed.  Rolling back package changes for 'PC01'.
StreamJsonRpc.RemoteInvocationException:  Package restore failed. Rolling back package changes for 'PC01'. (plus callstack)
```

This `RemoteInvocationException` contains some of the original information from the service, but not all of it; original NuGet exceptions are no longer propagated intact from the service to the client.  There is a bug in the VS SDK which causes the error handling experience to be different between local and remote scenarios.  When that bug is fixed, I can handle `RemoteInvocationException`s consistently in local and remote scenarios during package operations in PM UI.

**Comment by rrelyea** [comment]
fix shipped in 5.8
